### PR TITLE
Process brooklin.initializers in blueprint-composer model and introdu…

### PR DESCRIPTION
**Pull request by algairim**
…ce requirements

Signed-off-by: Mykola Mandra <mykola.mandra@cloudsoftcorp.com>

### CODE REVIEW

File: ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.template.html [hunk 1]
```diff
@@ -579,6 +579,12 @@ <h4>No matching configuration</h4>
     </div>
 </br-collapsible>
 
+<!-- ENTITY REQUIREMENTS -->
+<ng-include src="'blueprint-composer/component/spec-editor/section-requirements.html'"></ng-include>
+<script type="text/ng-template" id="blueprint-composer/component/spec-editor/section-requirements.html" defer-to-preexisting-id="true">
+    <!-- Placeholder for 'requirements' section. -->
+</script>
```
Comment (algairim) [inline ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.template.html:586]: This is the only functional change in the template.

File: ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.template.html [hunk 2]
```diff
@@ -688,9 +694,9 @@ <h4 ng-if="model.hasEnrichers() && filteredEnrichers.length === 0">No enrichers
 
 <ng-include src="'blueprint-composer/component/spec-editor/section-others.html'"></ng-include>
 <script type="text/ng-template" id="blueprint-composer/component/spec-editor/section-others.html" defer-to-preexisting-id="true">
+    <!-- Placeholder for other sections. -->
 </script>
```
Comment (algairim) [inline ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.template.html:698]: I would re-use 'others' section for requirements, however, it is not in the desired position. We agreed to have requirements displayed right under configuration.

**Review by duncangrant — APPROVED** [review]
LGTM thanks @algairim

File: ui-modules/blueprint-composer/app/components/util/model/entity.model.js [hunk 3]
```diff
@@ -1138,6 +1156,19 @@ function setParametersFromJson(incomingModel) {
...
 }
 
+/**
+ * Set initializers from JSON.
+ *
+ * @param {Object} model
+ * @param {Array} initializers
+ */
+function setInitializersFromJson(initializers) {
+    if (!Array.isArray(initializers)) {
```
Comment (duncangrant) [inline ui-modules/blueprint-composer/app/components/util/model/entity.model.js:1166]: I'm not sure throwing en Error here is right.  Should we be more tolerant of an incorrect type or null value here?

Comment (algairim) [reply ui-modules/blueprint-composer/app/components/util/model/entity.model.js:1166]: I followed the pattern handling errors in this model. Brooklyn initialisers is expected to be an array.

**Pull request merged**